File: app/src/main/java/zone/com/zadapter3/adapter/RightDiffDelegates.java
package zone.com.zadapter3.adapter;

import com.zone.adapter3.bean.Holder;
import com.zone.adapter3.bean.ViewDelegates;

import zone.com.zadapter3.R;

/**
 * [2017] by Zone
 */

public class RightDiffDelegates extends ViewDelegates<String> {


    @Override
    public int getLayoutId() {
        return R.layout.item_right;
    }

    @Override
    public void fillData(int postion, String data, Holder holder) {
        if (!"ChangePayload".equals(holder.getPayloads()))
            holder.setText(R.id.tv, data);
    }

    @Override
    public boolean isFullspan() {
        return false;
    }
}

File: app/src/main/java/zone/com/zadapter3kt/adapter/RightDiffDelegates.kt
package zone.com.zadapter3kt.adapter

import com.zone.adapter3kt.data.DataWarp

import zone.com.zadapter3.R
import zone.com.zadapter3kt.adapterimpl.HolderExDemoImpl
import zone.com.zadapter3kt.adapterimpl.ViewDelegatesDemo

/**
 * [2017] by Zone
 */

class RightDiffDelegates : ViewDelegatesDemo<String>() {
    override var layoutId: Int = R.layout.item_right


    override fun onBindViewHolder(position: Int, item: DataWarp<String>, baseHolder: HolderExDemoImpl, payloads: List<*>) {
        if (payloads.firstOrNull()?.equals("ChangePayload") ?: false)
            baseHolder.setText(R.id.tv, item.data!!)
    }

}
